plugins/outputs/influxdb_v2/http.go

```diff
@@ -166,7 +183,33 @@ func (c *httpClient) Write(ctx context.Context, metrics []telegraf.Metric) error
 		return nil
 	}
 
-	desc := resp.Header.Get("X-Influx-Error")
+	writeResp := &genericRespError{}
+	json.NewDecoder(resp.Body).Decode(writeResp)
+	var desc string
+
+	switch resp.StatusCode {
+	case http.StatusBadRequest: // 400
+		// LineProtocolError
+		desc = fmt.Sprintf("%s - %s;%d;%s", writeResp, writeResp.Op, writeResp.Line, writeResp.Err)
+	case http.StatusUnauthorized, http.StatusForbidden: // 401, 403
+		// Error
+		desc = fmt.Sprintf("%s - %s;%s", writeResp, writeResp.Op, writeResp.Err)
+	case http.StatusRequestEntityTooLarge: // 413
+		// LineProtocolLengthError
+		desc = fmt.Sprintf("%s - %s;%d", writeResp, writeResp.Op, writeResp.MaxLength)
+	case http.StatusTooManyRequests, http.StatusServiceUnavailable: // 429, 503
+		retryAfter := resp.Header.Get("Retry-After")
+		retry, err := strconv.Atoi(retryAfter)
+		if err != nil {
+			return fmt.Errorf("Bad value for 'Retry-After': %s", err.Error())
+		}
+		time.Sleep(time.Second * time.Duration(retry))
```

Comment:
You may want a max sleep just in case the server says you sleep for a long time.

Reply:
10s seems reasonable to me as a max sleep.  The big issue with the influxdb output is that, when under load, the next filled batch will almost immediately trigger the next write even after a write timeout.

Reply:
Oh but don't sleep and write because of #2919, hang on to the time and return.  On the next call to Write check the time and if the amount of time has not passed then return an error.

Let's define a special error type specifically for this, something like CircuitOpenErr?  We can put it in the internal package for now.

Reply:
Does the failed to write metric get cached, timestamp included? If the next call to write is less than the initial retry-time, would that second metric also get cached and be written on the next attempt that is after the retry-time? If the next write also returns a retry-time, is it acceptable to drop that new metric, even though it didn't get a chance to retry?
I put in a max retries limit based on the connection/`httpClient`, but it seemed wrong, would that be useful to put back in?